src/Avalonia.Base/ValueStore.cs

```diff
@@ -162,7 +162,7 @@ public void ClearLocalValue<T>(StyledPropertyBase<T> property)
                         _sink.ValueChanged(new AvaloniaPropertyChangedEventArgs<T>(
                             _owner,
                             property,
-                            old,
+                            new Optional<T>(old),
```

Comment:
This fixes the warning without using `!`. The warning is actually caused by a bug in Roslyn: https://github.com/dotnet/roslyn/issues/43225